app/main/helpers/frameworks.py

```diff
@@ -425,3 +428,46 @@ def validator(self, *args, **kwargs):
                                   "company details.")
 
         return True
+
+
+def return_404_if_applications_closed(data_api_client_callable):
+    def real_decorator(func):
+        @wraps(func)
+        def decorated_view(*args, **kwargs):
+            data_api_client = data_api_client_callable()
+            if 'framework_slug' not in kwargs:
+                current_app.logger.error("Required parameter `framework_slug` is undefined for the calling view.")
+                abort(500, "There was a problem accessing this page of your application. Please try again later.")
+
+            framework_slug = kwargs['framework_slug']
+            framework = get_framework_or_404(data_api_client, framework_slug)
+
+            if not framework['status'] == 'open':
+                current_app.logger.info(
+                    'Supplier {supplierId} requested "{method} {url}" after {frameworkSlug} applications closed.',
+                    extra={
+                        'supplierId': current_user.supplier_id,
```

Comment:
I *think* most people don't bother camel-casing `extra` parameters from python (though I understand why you are) - so actually these *could* end up mis-fitting with other places where people will use e.g. `supplier_id`. But it's not like it's important.

Reply:
Just looked, and it's snake case everywhere else. Will change.